### ISSUE TITLE
Windows CI checks fail due to timeout. 

### ISSUE THREAD

**Issue by fangfufu** [gsliepen/tinc#306]
The Windows CI checks (``windows`` and ``windows --disable-legacy-protocol``) have been failing due to exceeding the maximum execution time of 20 minutes. 

I had a look around, I couldn't find an issue or a pull request dedicated to this. So I am opening a new issue. 

Tagging @hg

**Comment by hg** [comment]
TBH I have no idea at the moment what the issue might be. See:

https://github.com/gsliepen/tinc/commit/e050753d1fc4d4465032919746c2a4db78f19932#comments

**Comment by fangfufu** [comment]
I am just wondering if we should turn the Windows CI workflow off for now. They aren't doing much right now. If we turn them off, we can get green ticks. 

What do you think, @gsliepen?

**Comment by fangfufu** [comment]
> TBH I have no idea at the moment what the issue might be. See:
> 
> [e050753#comments](https://github.com/gsliepen/tinc/commit/e050753d1fc4d4465032919746c2a4db78f19932#comments)

I can confirm this causes the problem, have a look at the run result of https://github.com/fangfufu/tinc/commit/5b3ed1540f385e708eee9ff025bfdd52f80665a1. I am not sure why it causes the problem either. I will try and investigate further.

**Comment by fangfufu** [comment]
I managed to replicate the same problem by running the CI script locally on my Windows 10 machine. I have no idea what might have caused it either. This is a bit beyond me.

**Comment by fangfufu** [comment]
@gsliepen, @hg, if you run the whole CI script under mingw32 toolchain, it works fine. So this problem is compiler specific.

**Comment by hg** [comment]
Yeah, this particular issue is triggered by `scripts.test`. `bar` repeatedly sends an Ethernet frame to `foo` with the type field set to zero. See this with wireshark:

[dump.zip](https://github.com/gsliepen/tinc/files/6973857/dump.zip)

It's easy to un-hang by assigning this field as is done elsewhere in the project:

https://github.com/hg/tinc/commit/fa7e7f42e642ad54f1c8b633506fbc5769c71139

the test suite passes just fine

- https://github.com/hg/tinc/runs/3301782019
- https://github.com/hg/tinc/runs/3301518716

but then hangs anyway after running the last test.

I'll dig into it later, I need to finish some other things first.

**Comment by fangfufu** [comment]
I managed to run this under MSYS2 MINGW32. This is my test folder. Do let me know if you want more things.

[test.zip](https://github.com/gsliepen/tinc/files/6975522/test.zip)

I tried to get Github to run the tests under MINGW32. It still hangs (https://github.com/fangfufu/tinc/commit/eb1faf3130f8c020dc257aa1242489dd3951f963), changing the time limit to 40 minutes doesn't help either (https://github.com/fangfufu/tinc/commit/9cd2fa369279cf7bf18e11d7c436fb21027f896c)

Now this behaviour seems to be Github only, which is really odd. However, I still think it is better to fix the code, rather than specifying the compiler / environment.
